src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/SingleMethodCompilationModuleGroup.cs

```diff
@@ -40,5 +41,23 @@ public override bool GeneratesPInvoke(MethodDesc method)
         {
             return true;
         }
+
+        public override bool TryGetModuleTokenForExternalType(TypeDesc type, out ModuleToken token)
```

Comment:
What is the SingleMethodCompilationModuleGroup.cs  good for?

Reply:
This is a debug aid for devs. If you have a compiler bug and it crashes compiling method X, you can pass a command line `--singlemethodname X` and it'll just compile the thing you care about.

Reply:
Yes, please implement this. Being able to compile a single method and run it through the compiler all the way to generating a PE file is quite useful.

---
In reply to: [357427440](https://github.com/dotnet/runtime/pull/816#discussion_r357427440) [](ancestors = 357427440)

Reply:
Done